2024-07-11 12:32:55.435344: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:32:55.435555: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:33:03,810 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:33:03,918 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:33:03,919 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:33:03,919 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:33:05,362 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:33:05,362 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:05,377 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.41}
INFO flwr 2024-07-11 12:33:05,377 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:33:05,579 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:33:05,601 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:33:05,604 | fedavg.py:242 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:05,610 | server.py:131 | fit progress: (1, 0.42652757357802634, {'accuracy': 0.79}, 0.23320740000000129)
DEBUG flwr 2024-07-11 12:33:05,611 | server.py:178 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:33:05,632 | server.py:192 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2024-07-11 12:33:05,632 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:33:05,632 | server.py:227 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:33:05,653 | server.py:241 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:05,663 | server.py:131 | fit progress: (2, 0.4216153532267744, {'accuracy': 0.8033333333333333}, 0.28589540000000113)
DEBUG flwr 2024-07-11 12:33:05,664 | server.py:178 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:33:05,681 | server.py:192 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2024-07-11 12:33:05,681 | server.py:227 | fit_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:33:05,702 | server.py:241 | fit_round 3 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:05,713 | server.py:131 | fit progress: (3, 0.42163877272390304, {'accuracy': 0.8033333333333333}, 0.33576730000000055)
DEBUG flwr 2024-07-11 12:33:05,714 | server.py:178 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:33:05,735 | server.py:192 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:33:05,735 | server.py:227 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:33:05,760 | server.py:241 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:05,773 | server.py:131 | fit progress: (4, 0.42349049415904033, {'accuracy': 0.79}, 0.3955180000000009)
DEBUG flwr 2024-07-11 12:33:05,774 | server.py:178 | evaluate_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:33:05,796 | server.py:192 | evaluate_round 4 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:33:05,796 | server.py:227 | fit_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:33:05,826 | server.py:241 | fit_round 5 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:05,838 | server.py:131 | fit progress: (5, 0.42264691919888736, {'accuracy': 0.79}, 0.4596264000000012)
DEBUG flwr 2024-07-11 12:33:05,838 | server.py:178 | evaluate_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:33:05,860 | server.py:192 | evaluate_round 5 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:33:05,860 | server.py:227 | fit_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:33:05,891 | server.py:241 | fit_round 6 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:05,902 | server.py:131 | fit progress: (6, 0.42263645201625927, {'accuracy': 0.79}, 0.5245953000000014)
DEBUG flwr 2024-07-11 12:33:05,902 | server.py:178 | evaluate_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:33:05,927 | server.py:192 | evaluate_round 6 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:33:05,927 | server.py:227 | fit_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:33:05,962 | server.py:241 | fit_round 7 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:05,975 | server.py:131 | fit progress: (7, 0.42263646816472306, {'accuracy': 0.79}, 0.5974873000000009)
DEBUG flwr 2024-07-11 12:33:05,976 | server.py:178 | evaluate_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:33:05,999 | server.py:192 | evaluate_round 7 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:33:06,000 | server.py:227 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:33:06,043 | server.py:241 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:06,057 | server.py:131 | fit progress: (8, 0.4223733081449492, {'accuracy': 0.79}, 0.678987900000001)
DEBUG flwr 2024-07-11 12:33:06,058 | server.py:178 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:33:06,083 | server.py:192 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:33:06,084 | server.py:227 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:33:06,134 | server.py:241 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:06,148 | server.py:131 | fit progress: (9, 0.42237543635345526, {'accuracy': 0.79}, 0.7704967000000007)
DEBUG flwr 2024-07-11 12:33:06,148 | server.py:178 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:33:06,179 | server.py:192 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:33:06,179 | server.py:227 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:33:06,219 | server.py:241 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:06,232 | server.py:131 | fit progress: (10, 0.4223754286274886, {'accuracy': 0.79}, 0.8542427000000004)
DEBUG flwr 2024-07-11 12:33:06,233 | server.py:178 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:33:06,260 | server.py:192 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2024-07-11 12:33:06,260 | server.py:153 | FL finished in 0.8820058000000017
INFO flwr 2024-07-11 12:33:06,261 | app.py:225 | app_fit: losses_distributed [(1, 0.4162844121456146), (2, 0.41590727493166924), (3, 0.4170657515525818), (4, 0.41647974848747255), (5, 0.42020766082264127), (6, 0.42020679229781743), (7, 0.420206823519298), (8, 0.42024445874350413), (9, 0.420244425535202), (10, 0.42024442851543425)]
INFO flwr 2024-07-11 12:33:06,261 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:33:06,261 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:33:06,261 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.42652757357802634), (2, 0.4216153532267744), (3, 0.42163877272390304), (4, 0.42349049415904033), (5, 0.42264691919888736), (6, 0.42263645201625927), (7, 0.42263646816472306), (8, 0.4223733081449492), (9, 0.42237543635345526), (10, 0.4223754286274886)]
INFO flwr 2024-07-11 12:33:06,261 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.41), (1, 0.79), (2, 0.8033333333333333), (3, 0.8033333333333333), (4, 0.79), (5, 0.79), (6, 0.79), (7, 0.79), (8, 0.79), (9, 0.79), (10, 0.79)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.90991746,  1.04518265,  0.78762441,  0.26600828,  0.1580615 ,
         0.04625279, -0.05066652,  0.04074236]]), array([0.02301435])]
[array([[-1.90991746,  1.04518265,  0.78762441,  0.26600828,  0.1580615 ,
         0.04625279, -0.05066652,  0.04074236]]), array([0.02301435])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.87861208,  1.05749928,  0.80213789,  0.30529817,  0.17353925,
         0.02206793, -0.05035863,  0.06655753]]), array([-0.03809643])]
[array([[-1.87861208,  1.05749928,  0.80213789,  0.30529817,  0.17353925,
         0.02206793, -0.05035863,  0.06655753]]), array([-0.03809643])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.88201836,  1.0590236 ,  0.80352207,  0.30567208,  0.17440045,
         0.02146091, -0.04955496,  0.06740201]]), array([-0.0403947])]
[array([[-1.88201836,  1.0590236 ,  0.80352207,  0.30567208,  0.17440045,
         0.02146091, -0.04955496,  0.06740201]]), array([-0.0403947])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.90617462,  0.9906362 ,  0.82796753,  0.31196801,  0.20504489,
         0.04115855, -0.05136257,  0.08104818]]), array([-0.05424894])]
[array([[-1.90617462,  0.9906362 ,  0.82796753,  0.31196801,  0.20504489,
         0.04115855, -0.05136257,  0.08104818]]), array([-0.05424894])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.88005858,  0.97981941,  0.80976822,  0.34839948,  0.2034959 ,
         0.04026266, -0.05748464,  0.0646423 ]]), array([-0.05977749])]
[array([[-1.88005858,  0.97981941,  0.80976822,  0.34839948,  0.2034959 ,
         0.04026266, -0.05748464,  0.0646423 ]]), array([-0.05977749])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.88043167,  0.98012985,  0.80991298,  0.34844339,  0.20365719,
         0.0402895 , -0.05755112,  0.06491502]]), array([-0.06013064])]
[array([[-1.88043167,  0.98012985,  0.80991298,  0.34844339,  0.20365719,
         0.0402895 , -0.05755112,  0.06491502]]), array([-0.06013064])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.88040678,  0.98012724,  0.80990583,  0.34843464,  0.20366   ,
         0.04030741, -0.05753501,  0.06492201]]), array([-0.06018402])]
[array([[-1.88040678,  0.98012724,  0.80990583,  0.34843464,  0.20366   ,
         0.04030741, -0.05753501,  0.06492201]]), array([-0.06018402])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.890069  ,  0.96023604,  0.81167657,  0.38602785,  0.18078693,
         0.02386884, -0.0321923 ,  0.05729331]]), array([-0.03550089])]
[array([[-1.890069  ,  0.96023604,  0.81167657,  0.38602785,  0.18078693,
         0.02386884, -0.0321923 ,  0.05729331]]), array([-0.03550089])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89007962,  0.96026479,  0.81166389,  0.38593922,  0.18082944,
         0.02387831, -0.03213032,  0.05726335]]), array([-0.03552755])]
[array([[-1.89007962,  0.96026479,  0.81166389,  0.38593922,  0.18082944,
         0.02387831, -0.03213032,  0.05726335]]), array([-0.03552755])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89007973,  0.96026477,  0.81166385,  0.38593939,  0.18082955,
         0.02387839, -0.03213031,  0.05726331]]), array([-0.03552742])]
[array([[-1.89007973,  0.96026477,  0.81166385,  0.38593939,  0.18082955,
         0.02387839, -0.03213031,  0.05726331]]), array([-0.03552742])]
>>>>>>>>>>>>>run time:
0.71875s
